resources/apt/apt.dsc.resource.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+export exist=true
+export NONINTERACTIVE=1
+
+# $packageName and $_exist are sent as env vars by dsc converting the JSON input to name/value pairs
+
+check_args() {
+    if [[ -z $packageName ]]; then
+        echo "packageName not set"
+        exit 1
+    fi
+}
+
+get_apt() {
+    pkgname=$1
+    InstalledSection=0
+    apt list --installed $pkgname 2>&1 | while read line; do
+        if [[ $line == Listing* ]]; then
+            InstalledSection=1
+        elif [[ $InstalledSection = 1 ]]; then
+            echo $line | awk '{
+                split($0, a, " ");
+		split(a[1], pn, "/");
+                printf("{ \"_exist\": \"%s\", \"packageName\": \"%s\", \"Version\": \"%s\", \"Source\": \"%s\" }\n", ENVIRON["exist"], pn[1], a[2], pn[2]);
+            }'
+        fi
+    done
+}
+
+if [[ "$#" -eq "0" ]]; then
+    echo "Command not provided, valid commands: get, set, export"
+    exit 1
+elif [[ "$1" == "get" ]]; then
+    check_args
+    output="$(get_apt $packageName)"
+    if [[ -z $output ]]; then
+	    printf '{ "_exist": "false", "packageName": "%s", "Version": "", "Source": "" }\n' $packageName
+    else
+	    echo $output
+    fi
+elif [[ "$1" == "set" ]]; then
+    check_args
+    if [[ -z $_exist ]]; then
+        # if $_exist is not defined in the input, it defaults to `true`
+        _exist=true
+    fi
+    if [[ $_exist = true ]]; then
+        apt install -y "${packageName}"
```

Comment:
Should this handle installing a specific version, since we have the `version` property? Or should `version` be marked as `readOnly`?

Reply:
I think this resource should support specifying the version, but limit it to apt syntax

Reply:
We can add this later